src/foam/java/Constant.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.java',
+  name: 'Constant',
+
+  properties: [
+    'name',
+    'javaType',
+    'value',
+    'javaValue',
+    'documentation'
+  ],
+
+  methods: [
+    function outputJava(o) {
+      o.indent();
+
+      if ( this.documentation != undefined ) {
+        o.out('/* ', this.documentation, ' */\n');
+        o.indent();
+        o.indent();
+      }
+
+      o.out('public static final ');
+
+      if ( this.type == undefined ) {
```

Comment:
what would you mean by that? As in we do not need to check if the type is undefined?